### DISCUSSION TITLE
CMCC XR30 好香，请大佬支持

### DISCUSSION THREAD

**Discussion by liyiranlab** [immortalwrt/immortalwrt#1339] [General]
CMCC XR30 好香，请大佬支持

**Comment by liyiranlab** [comment]
找到一些资料：
https://github.com/lgs2007m/Actions-OpenWrt/blob/main/%E5%88%B7%E6%9C%BA%E6%95%99%E7%A8%8B/RAX3000M%E7%AE%97%E5%8A%9B%E7%89%88%2BRAX3000Z%E5%A2%9E%E5%BC%BA%E7%89%88%E5%8D%95%E5%88%86%E5%8C%BA%E5%88%B7%E6%9C%BA%2BTTL%E6%95%91%E7%A0%96%E6%95%99%E7%A8%8B20240405.md

**Reply by zhuoking** [reply]
我直接immortalwrt downloads 上下载xay3000m 的na版刷成230.5 成功使用

**Comment by liyiranlab** [comment]
请问大佬，在源码中查到了5F1GM7UE GD的flash的代码，是否说明7981的机器都支持？能否直接刷机？或者需要代码中哪些部分添加？

**Reply by liyiranlab** [reply]
我入了一台 XR30 ，目前刷的h大的rax3000m固件，uboot是网上 玩数码的阿三 出的。尝试刷immortalwrt主线失败。有uboot启动信息。会适配的大佬看看，指点一下?

**Reply by liyiranlab** [reply]
NOTICE:  BL2: v2.10.0  (release):ImmortalWrt v2024.01.17~bacca82a-2 (mt7981-spim-nand-ddr4)
NOTICE:  BL2: Built : 11:16:17, Jul 10 2024
NOTICE:  WDT: Cold boot
NOTICE:  WDT: disabled
NOTICE:  EMI: Using DDR4 settings
NOTICE:  EMI: Detected DRAM size: 512MB
NOTICE:  EMI: complex R/W mem test passed
NOTICE:  CPU: MT7981 (1300MHz)
NOTICE:  SPI_NAND parses attributes from parameter page.
NOTICE:  SPI_NAND Detected ID 0xc8
NOTICE:  Page size 2048, Block size 131072, size 134217728
NOTICE:  BL2: Booting BL31
NOTICE:  BL31: v2.10.0  (release):ImmortalWrt v2024.01.17~bacca82a-2 (mt7981-spim-nand-ddr4)
NOTICE:  BL31: Built : 11:16:17, Jul 10 2024


U-Boot 2024.01-ImmortalWrt-r31033-97ca0819e9 (Jul 10 2024 - 11:16:17 +0000)

CPU:   MediaTek MT7981
Model: CMCC RAX3000M
DRAM:  512 MiB
Core:  36 devices, 18 uclasses, devicetree: embed
Loading Environment from UBI... spi-nand: spi_nand spi_nand@0: GigaDevice SPI NAND was found.
spi-nand: spi_nand spi_nand@0: 128 MiB, block size: 128 KiB, page size: 2048, OOB size: 128
ubi0 error: ubi_read_volume_table: the layout volume was not found
ubi0 error: ubi_attach_mtd_dev: failed to attach mtd5, error -22
UBI error: cannot attach mtd5
UBI error: cannot initialize UBI, error -22
UBI init error 22
Please check, if the correct MTD partition is used (size big enough?)

 Cannot find mtd partition "ubi"
In:    serial@11002000
Out:   serial@11002000
Err:   serial@11002000
reset button found
Loading Environment from UBI... ubi0 error: ubi_read_volume_table: the layout volume was not found
ubi0 error: ubi_attach_mtd_dev: failed to attach mtd5, error -22
UBI error: cannot attach mtd5
UBI error: cannot initialize UBI, error -22
UBI init error 22
Please check, if the correct MTD partition is used (size big enough?)

 Cannot find mtd partition "ubi"
Net:   
Warning: ethernet@15100000 (eth0) using random MAC address - 6e:34:db:4e:63:88
eth0: ethernet@15100000
 7 [r [999;999H [6n 8No EFI system partition
No EFI system partition
Failed to persist EFI variables
Error, no UBI device selected!
Error, no UBI device selected!
Erasing 0x00000000 ... 0x071fffff (912 eraseblock(s))
resetting ...


F0: 102B 0000

FA: 1040 0000

FA: 1040 0000 [0200]

F9: 0000 0000

V0: 0000 0000 [0001]

00: 0000 0000

BP: 2400 0041 [0000]

G0: 1190 0000

EC: 0000 0000 [1000]

T0: 0000 0236 [010F]

Jump to BL


NOTICE:  BL2: v2.10.0  (release):ImmortalWrt v2024.01.17~bacca82a-2 (mt7981-spim-nand-ddr4)
NOTICE:  BL2: Built : 11:16:17, Jul 10 2024
NOTICE:  WDT: [40000000] Software reset (reboot)
NOTICE:  EMI: Using DDR4 settings
NOTICE:  EMI: Detected DRAM size: 512MB
NOTICE:  EMI: complex R/W mem test passed
NOTICE:  CPU: MT7981 (1300MHz)
NOTICE:  SPI_NAND parses attributes from parameter page.
NOTICE:  SPI_NAND Detected ID 0xc8
NOTICE:  Page size 2048, Block size 131072, size 134217728
NOTICE:  BL2: Booting BL31
NOTICE:  BL31: v2.10.0  (release):ImmortalWrt v2024.01.17~bacca82a-2 (mt7981-spim-nand-ddr4)
NOTICE:  BL31: Built : 11:16:17, Jul 10 2024

**Reply by liyiranlab** [reply]
上面，刷了主线bl2 和 fip ，后砖了，ttl信息。

**Comment by cncherisher** [comment]
应该是需要按照h大这个提交https://github.com/hanwckf/bl-mt798x/commit/5ba6fa3783e3d299367504cbbe0732f1b9b494fc
去修改uboot中nand的驱动
https://github.com/immortalwrt/immortalwrt/blob/f6bd6b31325891d9c61b64b36a0acd7192bbd7b6/package/boot/uboot-mediatek/patches/100-22-mtd-spi-nand-backport-from-upstream-kernel.patch#L707
然后再自己编译bl2和fip

**Comment by asianbigcock** [comment]
看这个 patch 的 commit,说明 xr30 已经可以直接刷 rax3000m nand 版 immortalwrt 了( =>24.10.3 ) https://github.com/immortalwrt/immortalwrt/commit/b66313ba30f87be06541cacfda2027564aeaddf4

**Reply by sealandwu** [reply]
大佬能简要说明下如何刷吗？ Uboot和bl2各用哪个版本的？

> 看这个 patch 的 commit,说明 xr30 已经可以直接刷 rax3000m nand 版 immortalwrt 了( =>24.10.3 ) [b66313b](https://github.com/immortalwrt/immortalwrt/commit/b66313ba30f87be06541cacfda2027564aeaddf4)

**Reply by asianbigcock** [reply]
> 大佬能简要说明下如何刷吗？ Uboot和bl2各用哪个版本的？
> 
> > 看这个 patch 的 commit,说明 xr30 已经可以直接刷 rax3000m nand 版 immortalwrt 了( =>24.10.3 ) [b66313b](https://github.com/immortalwrt/immortalwrt/commit/b66313ba30f87be06541cacfda2027564aeaddf4)

https://downloads.immortalwrt.org/releases/24.10.4/targets/mediatek/filogic/immortalwrt-24.10.4-mediatek-filogic-cmcc_rax3000m-nand-preloader.bin
https://downloads.immortalwrt.org/releases/24.10.4/targets/mediatek/filogic/immortalwrt-24.10.4-mediatek-filogic-cmcc_rax3000m-nand-bl31-uboot.fip
https://downloads.immortalwrt.org/releases/24.10.4/targets/mediatek/filogic/immortalwrt-24.10.4-mediatek-filogic-cmcc_rax3000m-initramfs-recovery.itb
https://downloads.immortalwrt.org/releases/24.10.4/targets/mediatek/filogic/immortalwrt-24.10.4-mediatek-filogic-cmcc_rax3000m-squashfs-sysupgrade.itb

24.10.4固件已经包含最新patch,如果不熟悉tftp刷入方式不建议刷.
